src/styles/withStyles.d.ts

```diff
@@ -28,19 +30,35 @@ export interface WithStylesOptions {
 
 export type ClassNameMap<ClassKey extends string = string> = Record<ClassKey, string>;
 
-export interface WithStyles<ClassKey extends string = string> {
+export interface WithStyles<ClassKey extends string = string> extends Partial<WithTheme> {
   classes: ClassNameMap<ClassKey>;
-  theme?: Theme;
 }
 
 export interface StyledComponentProps<ClassKey extends string = string> {
   classes?: Partial<ClassNameMap<ClassKey>>;
   innerRef?: React.Ref<any>;
 }
 
+/*
+  The Omit<P, 'classes'> below (rather than the canonical Omit<P, WithStyles<ClassKey>>)
+  is used to allow interoperability with withTheme, which may provide the 'theme' prop.
+ */
+/*
+export default function withStyles<ClassKey extends string>(
+  style: StyleRules<ClassKey> | StyleRulesCallback<ClassKey>,
+  options?: WithStylesOptions,
+): <P extends WithStyles<ClassKey>>(
+  component: React.ComponentType<P>,
+) => React.ComponentType<Omit<P, 'classes'> & StyledComponentProps<ClassKey>>;
+*/
+
+interface MaybeWithStyles<ClassKey extends string> extends Partial<WithStyles<ClassKey>> {
+  [k: string]: any;
+}
```

Comment:
How about making a reusable version of this,

```ts
type ConsistentWith<O> = Partial<O> & Record<string, any>;
```

which could then be used for `withStyles`, `withWidth` and `withTheme`:

```ts
P extends ConsistentWith<WithStyles<ClassKey>>
P extends ConsistentWith<WithWidth>
P extends ConsistentWith<WithTheme>
```